### PR TITLE
Alerting/Chore: Fix TimeRangeInput not working across multiple months

### PR DESCRIPTION
**Pull request by tomratcliffe**
**What is this feature?**

Fixes the `TimeRangeInput` component not working across multiple months.

**Why do we need this feature?**

Without this, it's fiddly to select a date range across a month. 
i.e. if the start and end dates are on separate pages of the `react-calendar`, then the onChange logic/selection won't work properly.

To replicate before, follow the same steps as the unit test, but that boils down to the following in storybok

* Select 1st of the month
* Go forward 2 months
* Select any date

The picker will close without updating the date, instead of selecting the correct range

The use of `TimeRangeInput` in Alerting is one of only 2 that I could find - the other being in Dashboards somewhere.
Ideally, we would instead deprecate this component and instead generalise the `TimeRangePicker` to work with this as well - as they work very similarly.

**Who is this feature for?**

Alerting UI users, as this was causing a problem in Silences creation

**Which issue(s) does this PR fix?**:

Customer reported bug

### CODE REVIEW

File: packages/grafana-ui/src/components/DateTimePickers/TimeRangeInput.test.tsx
Comment (tomratcliffe) [inline packages/grafana-ui/src/components/DateTimePickers/TimeRangeInput.test.tsx:1]: This test works nicely, apart from the fact that it passes even without the fix D: Not sure why it still works with the original implementation, perhaps relating to `jest-dom` and the clickaway handler logic?

Any ideas?

File: packages/grafana-ui/src/components/DateTimePickers/TimeRangeInput.tsx [hunk 1]
```diff
@@ -77,13 +79,30 @@ export const TimeRangeInput = ({
     onChange({ from, to, raw: { from, to } });
   };
 
+  const overlayRef = createRef<HTMLElement>();
```
Comment (gillesdemey) [inline packages/grafana-ui/src/components/DateTimePickers/TimeRangeInput.tsx:82]: Why do we use `createRef` for the overlayRef but `useRef` for the button ref? 🤔

**Review by gillesdemey — APPROVED** [review]
Interaction does work as intended and I failed to replicate the bug report – LGTM

**Comment by gillesdemey** [comment]
For posterity; ideally we wouldn't use this component as its interaction is sort of confusing (the start and end date are separate fields but you choose both based on a single interaction with any one of them 😕).

If we can managed to replace the one other instance of this component outside of alerting we could mark this component as deprecated.

**Comment by tomratcliffe** [comment]
I tried to get the test to fail without the fix but wasn't able to - seems to be related to jest-dom and clicking outside the element etc.

File: packages/grafana-ui/src/components/DateTimePickers/TimeRangeInput.tsx (hunk 1 above)
Comment (tomratcliffe) [reply packages/grafana-ui/src/components/DateTimePickers/TimeRangeInput.tsx:82]: I don't know, I must have changed it accidentally when re-implementing from the other component 🤷 I'll change it to be consistent with `TimeRangePicker`, thanks for spotting it

**Pull request merged**